snap/validate.go

```diff
@@ -454,13 +460,31 @@ func validateAppOrderCycles(apps map[string]*AppInfo) error {
 	return nil
 }
 
+// Whitelist of external systemd targets that a service can use
+// for before/after ordering. This is useful for e.g. classic snaps
+// that need to run snapd itself or that need to run commands from
+// snaps that are in the seed so they can only start working once
+// the seeded has happened.
+var allowedExternalServices = []string{
+	"snapd",
+	"snapd.seeded",
+}
+
 func validateAppOrderNames(app *AppInfo, dependencies []string) error {
 	// we must be a service to request ordering
 	if len(dependencies) > 0 && !app.IsService() {
 		return fmt.Errorf("cannot define before/after in application %q as it's not a service", app.Name)
 	}
 
 	for _, dep := range dependencies {
+		if strings.HasPrefix(dep, "external:") {
+			shortName := strings.TrimPrefix(dep, "external:")
+			if !strutil.ListContains(allowedExternalServices, shortName) {
+				return fmt.Errorf("cannot use external name %q, only %s is allowed", shortName, strutil.Quoted(allowedExternalServices))
```

Comment:
Showing the full list of allowed values may get infeasible eventually but for now it is a nice touch. +1